packages/ui/src/ui/store/actions/system/schedulers.js

```diff
@@ -118,3 +122,34 @@ export function getAgents() {
             return getAgentsState(res);
         });
 }
+
+export const changeSchedulerMaintenance =
+    ({address, maintenance, message, type}) =>
+    async (dispatch, getSate) => {
+        const state = getSate();
+        const isAgent = type === 'agents';
+        const items = isAgent ? getSystemAgents(state) : getSystemSchedulers(state);
+
+        const master = items.find((i) => {
+            const host = i.host.$value;
+            const annotations = ypath.getValue(i.host, '/@annotations');
+            return host === address || annotations.physical_host === address;
```

Comment:
It is more strict to check only corresponding value according to current mode (`host`/`container`). Also it explicitly highlights dependencies in the code.